### PR TITLE
Fix facet filters for ranges

### PR DESCRIPTION
**Pull request by ckeboss**
## Description
Currently the ability to filter a Facet (`isFilterable` prop) does not work with range based facets for app-search, as the structure of the `value` property is different. Assuming this is intended, this change gives the ability to filter by that type of value.

If this is unintentional, we can change the connector to change the way the value structured in the response.

## List of changes
A bit of duck typing for the `value` property in `facetValues` to facilitate filtering better.

### CODE REVIEW

File: packages/react-search-ui/src/containers/Facet.tsx [hunk 1]
```diff
@@ -92,13 +92,30 @@ export class FacetContainer extends Component<
...
+        let valueToSearch;
+        switch (typeof option.value) {
+          case "string":
+            valueToSearch = accentFold(option.value).toLowerCase();
+            break;
+          case "number":
+            valueToSearch = option.value.toString();
+            break;
+          case "object":
+            valueToSearch =
```
Comment (joemcelroy) [inline packages/react-search-ui/src/containers/Facet.tsx:105]: nitpick: could you use optional chaining to improve readability? 

`typeof option?.value?.name === "string"`

**Review by joemcelroy — COMMENTED** [review]
Looks good other than small nitpick!

File: packages/react-search-ui/src/containers/Facet.tsx (hunk 1 above)
Comment (ckeboss) [reply packages/react-search-ui/src/containers/Facet.tsx:105]: Done

**Review by joemcelroy — APPROVED** [review]

**Comment by joemcelroy** [comment]
thanks for your submission as always!

**Comment by ckeboss** [comment]
@joemcelroy any idea when a new patch version will be published?

**Comment by joemcelroy** [comment]
will try today or tomorrow. Unfortunately its a manual process right now.

**Pull request merged**